### DISCUSSION TITLE
eks-aws-auth-merger not working

### DISCUSSION THREAD

**Discussion by DeepthiKodali2305** [gruntwork-io/knowledge-base#331] [Help] [answered]
eks-aws-auth-merger was deployed successfully but the additional nodes deployed which needed to be added to the cluster throw the below error

│ Error: Post "http://localhost/api/v1/namespaces/aws-auth-merger/configmaps": dial tcp 127.0.0.1:80: connect: connection refused

**Comment by autero1** [comment]
Hi, could you provide a bit more context so I can investigate what might be causing this? Where/when do you get this error, Terraform, container logs,..? Sometimes this sort of error comes when your `kubernetes` or `helm` provider isn't properly configured and Terraform tries to contact the localhost api server.

**Comment by DeepthiKodali2305** [comment]
Hi Team,

I am getting this error even when we run Terragrunt apply from the gitlab pipelines and also even from my local if I run terragrunt apply.

We have all the provider configuration below:

generate "k8s_provider" {
  path      = "k8s_provider.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
provider "kubernetes" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}


 required_providers {
    helm = {
      source  = "hashicorp/helm"
      version = ">= 2.4.0"
    }

required_providers {
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = ">= 2.0.0"
    }
  }
}

 required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.50.0"
    }

Thanks and regards
Phani Deepthi Kodali




C2 General

**Reply by autero1** [reply]
Ok, so it definitely looks like a provider configuration issue. How do you include the provider generate block in the hcl file you're trying to apply?

**Comment by DeepthiKodali2305** [comment]
Hi Petri,

I include the following lines of code in eks-aws-auth-merger terragrunt.hcl

generate "k8s_provider" {
  path      = "k8s_provider.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
provider "kubernetes" {
  experiments {
    manifest_resource = true
  }
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}





C2 General

**Comment by DeepthiKodali2305** [comment, marked as answer]
Hi Petri,

I have identified the issue.
Provider block was missing from managed nodes  hcl files and I have added them now

Thanks and regards
Phani Deepthi Kodali



C2 General
From: Kodali, Deepthi, Vodafone (External)
Sent: 04 April 2022 12:37
To: gruntwork-io/knowledge-base ***@***.***>; gruntwork-io/knowledge-base ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [gruntwork-io/knowledge-base] eks-aws-auth-merger not working (Discussion #331)

Hi Petri,

I include the following lines of code in eks-aws-auth-merger terragrunt.hcl

generate "k8s_provider" {
  path      = "k8s_provider.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
provider "kubernetes" {
  experiments {
    manifest_resource = true
  }
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}





C2 General
